File: code/1732.find-the-highest-altitude.php
<?php
/*
 * @lc app=leetcode id=1732 lang=php
 *
 * [1732] Find the Highest Altitude
 */

// @lc code=start
class Solution {

    /**
     * @param Integer[] $gain
     * @return Integer
     */
    function largestAltitude($gain) {
        $max = 0;
        $current = 0;
        foreach ($gain as $val) {
            $current += $val;
            $max = $current > $max ? $current : $max;
        }
        return $max;
    }
}
// @lc code=end

File: code/1732.find-the-highest-altitude.go
package code

/*
 * @lc app=leetcode id=1732 lang=golang
 *
 * [1732] Find the Highest Altitude
 */

// @lc code=start
func largestAltitude(gain []int) int {
	// prefixSum := make([]int, len(gain)+1)
	// for i := range gain {
	// 	prefixSum[i] = 0
	// }

	// for i, v := range gain {
	// 	prefixSum[i+1] = prefixSum[i] + v
	// }

	// return slices.Max(prefixSum)

	maxAltitude := 0
	currentAltitude := 0

	for _, v := range gain {
		currentAltitude += v
		if currentAltitude > maxAltitude {
			maxAltitude = currentAltitude
		}
	}
	return maxAltitude
}

// @lc code=end
